server/util.go

```diff
@@ -292,24 +292,16 @@ func dumpBinaryRow(buffer []byte, columns []*ColumnInfo, row chunk.Row, d *resul
 
 type inputDecoder struct {
 	encoding charset.Encoding
-
-	buffer []byte
```

Comment:
So the assumption is: on average, there are few parameters in prepared statements?

Reply:
no, we can not buffer here since there is a `types.NewBytesDatum` use the `decodeInput` result which will take the ownership, it means we must make a new `bytes array`